File: pyg_parent/pyg_sellergoods_interface/src/main/java/com/pyg/pyg_sellergoods_interface/ItemCatService.java
package com.pyg.pyg_sellergoods_interface;

import java.util.List;

import com.pyg_pojo.entity.PageResult;
import com.pyg_pojo.pojo.TbItem;
import com.pyg_pojo.pojo.TbItemCat;
import com.pyg_pojo.pojo.TbItemCatExample;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface ItemCatService {


    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbItemCat> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    public List<TbItemCat> findPage(Long id);


    /**
     * 增加
     */
    public void add(TbItemCat itemCat);


    /**
     * 修改
     */
    public void update(TbItemCat itemCat);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbItemCat findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);


    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    /*public PageResult findPage(TbItemCat itemCat, int pageNum, int pageSize);*/

}

File: pyg_parent/pyg_manager_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService', function ($http) {

    //查询全部商品分类信息
    this.findAll=function () {
        return $http.get('../itemCat/findAll.do');
    }


    //根据parent_id查询
    this.findNextMun = function (id) {
        return $http.get('../itemCat/findPage.do?id=' + id);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.get('../itemCat/findOne.do?id=' + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post('../itemCat/add.do', entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../itemCat/update.do', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../itemCat/delete.do?ids=' + ids);
    }

});
